bin/pyspark

```diff
@@ -33,8 +33,9 @@ export _SPARK_CMD_USAGE="Usage: ./bin/pyspark [options]"
 # PYSPARK_DRIVER_PYTHON_OPTS to pass options when starting the Python driver
 # (e.g. PYSPARK_DRIVER_PYTHON_OPTS='notebook').  This supports full customization of the IPython
 # and executor Python executables.
-#
-# For backwards-compatibility, we retain the old IPYTHON and IPYTHON_OPTS variables.
+# 
```

Comment:
The whole block of docs above this could be removed.
while you're at it, one last change would be helpful: mention in `programming-guide.md` that `jupyter` works too, where it shows the `ipython` example. Just a sentence noting that is OK.
LGTM otherwise.